include/metalldata/impl/metall_graph_topk.ipp

```diff
@@ -0,0 +1,131 @@
+#pragma once
+#include <metalldata/metall_graph.hpp>
+#include <functional>
+#include <queue>
+#include "ygm/detail/collective.hpp"
+#include "ygm/utility/assert.hpp"
+
+namespace metalldata {
+
+template <typename Compare>
+std::vector<std::vector<metall_graph::data_types>> metall_graph::topk(
+  size_t k, const series_name& ser_name,
+  const std::vector<series_name>& ser_inc, Compare comp,
+  const where_clause& where) {
+  record_store_type* pdata;
+  std::function<void(std::function<void(size_t)>, const where_clause&)>
+    for_all_func;
+  if (ser_name.is_edge_series()) {
+    pdata        = m_pedges;
+    for_all_func = [this](std::function<void(size_t)> func,
+                          const where_clause&         where) {
+      priv_for_all_edges(func, where);
+    };
+  } else if (ser_name.is_node_series()) {
+    pdata        = m_pnodes;
+    for_all_func = [this](std::function<void(size_t)> func,
+                          const where_clause&         where) {
+      priv_for_all_nodes(func, where);
+    };
+  } else {
+    return {};
+  }
+
+  if (!has_series(ser_name)) {
+    return {};
+  }
+
+  // we make sure that the compared column is element 0. This
+  // also guarantees that the vector is not empty.
+  std::vector<std::string> ser_inc_unq{std::string(ser_name.unqualified())};
+
+  for (const auto& ser : ser_inc) {
```

Comment:
Missing check to ensure that all series in ser_inc are of the same qualifier as ser_name.

Reply:
Added checks.